lib/QubitFlatfileImport.class.php

```diff
@@ -1510,27 +1515,63 @@ public function createOrUpdateNote($typeId, $text, $options = array())
   }
 
   /**
-   * Return whether a note already exists given specified parameters.
+   * Return an array containing all note content for the object, type and culture.
    *
-   * This function is to prevent creating duplicate notes when updating descriptions.
+   * This function is used to build a list to match against existing to prevent
+   * creating duplicate notes when updating descriptions.
    *
    * @param int $objectId  Object id for object that the note belongs to.
    * @param int $typeId  Note type id indicating note type.
-   * @param string $content  Note content to check against.
    * @param string $culture  Note culture to check against.
    *
-   * @return bool  True if the same note exists, false otherwise.
+   * @return bool  Array of all note content.
    */
-  private function checkNoteExists($objectId, $typeId, $content, $culture)
+  private function getExistingNotes($objectId, $typeId,  $culture)
   {
-    $query = "SELECT n.id FROM note n
+    $existingNotes = array();
+
+    $query = "SELECT n.id, i.content FROM note n
       INNER JOIN note_i18n i ON n.id=i.id
       WHERE n.object_id=?
       AND n.type_id=?
-      AND i.content=?
       AND i.culture=?";
-    $statement = self::sqlQuery($query, array($objectId, $typeId, $content, $culture));
-    return false !== $statement->fetch(PDO::FETCH_OBJ);
+
+    $statement = self::sqlQuery($query, array($objectId, $typeId,  $culture));
+
+    foreach ($statement->fetchAll(PDO::FETCH_OBJ) as $row)
+    {
+      $existingNotes[] = $row->content;
+    }
+    return $existingNotes;
+  }
+
+  /**
+   * Return whether a note already exists given specified parameters.
+   *
+   * This function is to prevent creating duplicate notes when updating descriptions.
+   *
+   * @param array $existingNotes  Notes already found in the database for this
+   * @param int $objectId  Object id for object that the note belongs to.
+   * @param int $typeId  Note type id indicating note type.
+   * @param string $content  Note content to check against.
+   * @param string $culture  Note culture to check against.
+   *
+   * @return bool  True if the same note exists, false otherwise.
+   */
+  private function checkNoteExists(&$existingNotes, $objectId, $typeId, $content, $culture)
```

Comment:
Only $existingNotes and $content seem used in the function body. Should the other parameters be removed?

Reply:
Hi @mcantelon!  Thanks for spotting that.  I have updated the function params in the new commit.

Reply:
Cool... looks good now.